src/main/java/kitchenpos/application/TableGroupService.java

```diff
@@ -76,15 +53,19 @@ public void ungroup(final Long tableGroupId) {
                 .map(OrderTable::getId)
                 .collect(Collectors.toList());
 
-        if (orderDao.existsByOrderTableIdInAndOrderStatusIn(
-                orderTableIds, Arrays.asList(OrderStatus.COOKING.name(), OrderStatus.MEAL.name()))) {
-            throw new IllegalArgumentException();
-        }
+        validateOrderStatus(orderDao.findAllByOrderTableIdIn(orderTableIds));
 
         for (final OrderTable orderTable : orderTables) {
-            orderTable.setTableGroupId(null);
-            orderTable.setEmpty(false);
+            orderTable.ungroup();
             orderTableDao.save(orderTable);
         }
     }
+
+    private void validateOrderStatus(final List<Order> orders) {
+        for (Order order : orders) {
```

Comment:
```suggestion
        for (final Order order : orders) {
```
여기다가도 final을 붙일 수 있답니다.